### ISSUE TITLE
Bug in flask-login example

### ISSUE THREAD

**Issue by mehaase** [pallets-eco/flask-principal#22]
The flask-login example in the documentation is extremely helpful (common use case, I would think) but it has a significant bug in it!

The sample code shows Flask-Principal being registered before Flask-Login, but if you do this, then flask principal will send the `identity_loaded` message _before_ Flask-Login has had a chance to load the current_user proxy. Therefore, the `identity_loaded` handler will always return the anonymous user!

I realize it's not meant to be a complete example, but it tripped me up as well as at least one person on Stack Overflow: http://stackoverflow.com/questions/18190067/flask-login-and-principal-current-user-is-anonymous-even-though-im-logged-in/18935921

It can be fixed simply by reversing the order of `Principal(app)` and `login_manager = LoginManager(app)`.

**Comment by joshfriend** [comment]
I was having trouble with this for a while as well. Because I don't want to use the session for user login in a REST api, I can't call `login_user` from Flask-Login because that tries to set a session cookie. Thus, the `current_user` is unset until after `load_user_from_request` returns so I can't call `principal.set_identity` from inside the request loader without triggering the recursive loop. I solved the issue by adding an intermediate signal callback to catch the `user_loaded_from_request` signal which sets the identity:


```python
@login_manager.request_loader
def load_user_from_request(request):
    user = get_user(request)
    return user

@user_loaded_from_request.connect
def on_user_loaded_from_request(sender, user):
    principal.set_identity(Identity(user.id))

@identity_loaded.connect
def on_identity_loaded(sender, identity):
    identity.user = current_user
    identity.provides.add(UserNeed(current_user.id))
    # Etc...
```

It would be nice if there was a way to configure Flask-Login to not use the session at all.

**Comment by bodgit** [comment]
Thanks @joshfriend for your example. I've been trying to tie Flask-Login/Flask-Principal together for a REST API yet I couldn't get the `on_identity_loaded` method to work if I attempt anything with `current_user`, I realise now this causes some sort of recursive loop. :+1:

**Comment by joshfriend** [comment]
> this causes some sort of recursive loop.

Yes. You **_cannot_** use `current_user` inside `login_manager.request_loader` because the value that the `current_user` proxy references is not set until a value is returned from the `request_loader`. Referencing `current_user` before it is set causes flask-login to try to load the user from the request by calling `request_loader` and... :boom: :boom: :boom:
